WalletWasabi.Fluent/Behaviors/AutoPasteAmountBehavior.cs

```diff
@@ -0,0 +1,84 @@
+using System.Globalization;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using Avalonia.Controls;
+using Avalonia.Controls.Primitives;
+using Avalonia.Input;
+using WalletWasabi.Fluent.Controls;
+using WalletWasabi.Fluent.Extensions;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.Fluent.Behaviors;
+
+public class AutoPasteAmountBehavior : AttachedToVisualTreeBehavior<DualCurrencyEntryBox>
+{
+	protected override void OnAttachedToVisualTree(CompositeDisposable disposable)
+	{
+		if (AssociatedObject is null)
+		{
+			return;
+		}
+
+		if (!Services.UiConfig.AutoPaste)
+		{
+			return;
+		}
+
+		AutoPaster(AssociatedObject, "PART_LeftEntryBox", IsValidBtc)
+			.DisposeWith(disposable);
+
+		AutoPaster(AssociatedObject, "PART_RightEntryBox", IsValidUsd)
+			.DisposeWith(disposable);
+	}
+
+	private static bool IsValidUsd(string s)
+	{
+		var canParse = decimal.TryParse(s, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture, out var d);
+		return canParse && CountDecimalPlaces(d) <= 2 && d > 0;
+	}
+
+	private static bool IsValidBtc(string s)
+	{
+		var canParseAsDecimal = decimal.TryParse(s, NumberStyles.AllowDecimalPoint, CultureInfo.InvariantCulture, out var d);
+
+		if (canParseAsDecimal)
+		{
+			var lessThanMaximum = d < Constants.MaximumNumberOfSatoshis;
+			var hasValidDecimalPlaces = CountDecimalPlaces(d) <= 8;
+			return lessThanMaximum && hasValidDecimalPlaces && d > 0;
+		}
+
+		return false;
+	}
+
+	private static decimal CountDecimalPlaces(decimal dec)
```

Comment:
IMO all this method is not needed. I tested without this and everything worked properly.

Reply:
Yes, it is needed. For example, 0.001 is a valid decimal, but it's not a valid USD amount.
Same for BTC: 0.0000000001 is a valid decimal, but it's not BTC.

Removing this will cause invalid behavior.

Reply:
0.0000000001 will be pasted as 0.00000000 which is correct. Any decimal place after 8 will be just ignored which makes sense.

0.001 USD will be pasted as 0.00 USD, and 0.009 will be pasted as 0.01 USD, which also makes sense.

Reply:
Let's discuss this in the VDG meeting.

Reply:
Was there a decision regarding this?

Reply:
Not sure, to be honest, but I would leave as is. For me, it's not correct to paste 0.001 as 0.00 USD. I'd preserve the original amount.

Anyways @zkSNACKs/visual-design-group please take a look and decide what to do.

Reply:
@SuperJMN Do not close this conversation until a decision is made on this.

Reply:
Discussed. Team decided the content detection to be "smart".